autoprecompiles/src/trace_handler.rs

```diff
@@ -0,0 +1,89 @@
+use itertools::Itertools;
+use rayon::prelude::*;
+use std::cmp::Eq;
+use std::collections::{BTreeMap, HashMap};
+use std::hash::Hash;
+
+pub struct TraceHandlerData<'a, F> {
+    pub dummy_values: Vec<Vec<&'a [F]>>,
+    pub dummy_trace_index_to_apc_index_by_instruction: Vec<HashMap<usize, usize>>,
+}
+
+pub trait TraceHandler {
+    type AirId: Hash + Eq + Sync;
+    type Field: Sync;
+
+    /// Returns a vector with the same length as original instructions
+    fn original_instruction_air_ids(&self) -> Vec<Self::AirId>;
+
+    fn original_instruction_subs(&self) -> Vec<Vec<u64>>;
```

Comment:
This is returned by apc generation, right? So the client would not need to implement this